clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -475,7 +472,7 @@ private void doSend(ClientRequest clientRequest, boolean isInternalRequest, long
 
         long metadataTimeout = metadataUpdater.maybeUpdate(now);
         try {
-            this.selector.poll(Utils.min(timeout, metadataTimeout, requestTimeoutMs));
+            this.selector.poll(Utils.min(timeout, metadataTimeout, minRequestTimeoutMs));
```

Comment:
I'm wondering if it still makes sense to consider metadata timeout and request timeout to determine the timeout for `selector.poll`, but maybe it can be discussed in another PR..